src/arbAggregatorPrepareTransactionRequest.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  PublicClient,
+  encodeFunctionData,
+  EncodeFunctionDataParameters,
+  Address,
+  Chain,
+  Transport,
+} from 'viem';
+
+import { arbAggregator } from './contracts';
+import { upgradeExecutorEncodeFunctionData } from './upgradeExecutor';
+import { Prettify } from './types/utils';
+
+type ArbAggregatorEncodeFunctionDataParameters = Prettify<
+  Omit<EncodeFunctionDataParameters<typeof arbAggregator.abi, string>, 'abi'>
+>;
```

Comment:
using `string` here means we won't infer `args` based on the function name. You can have a look at `arbOwnerPrepareTransactionRequest`

Reply:
Thanks Chris!
It looks like I copied the previous code when creating the decorator, and it definitely looks nicer now. I've made the necessary changes (waiting for the tests to finish)